### PR TITLE
added tests for Authorisations, Navigation to authorisation & Navigat…

### PR DESCRIPTION
**Pull request by VijayKumarTech**
…ion through authorisations

### CODE REVIEW

**Review by jackheslop96 — DISMISSED** [review]

**Review by JohnWhitfield — CHANGES_REQUESTED** [review]
AuthorisationsStepDef.scala
Line 47 authorisation needs an 's?' after the 'n'.
Line 55 authorisations needs a '?' after the 's'.
Line 70 needs 'the' inserted before 'date'
Line 23 I don't understand why the Arrivals/Identification/AuthorisationsReferenceNumberPage.scala has been changed as the ticket was for depatures frontend.
Remove 'An' from the file name of AddAnAuthorisationPage.scala
Remove 'An' from the file name of RemoveAnAuthorisationPage.scala
Remove 'For' from the file name of ReferenceNumberForAuthorisationPage.scala
There are no tests in either of the e2e features at all only the TransportDetails.feature.
In the TransportDetails.feature you have added a new scenario which would not be needed when the e2e tests are updated.

**Review by vajarova — APPROVED** [review]

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  And("""^(?:I )?enter (.+) for What is the reference number for the (.*) page$""") {
+    (answer: String, typeOfAuthorisation: String) =>
+      AuthorisationsReferenceNumberPage
+        .loadPage(typeOfAuthorisation)
+        .fillInput(answer)
+        .submitPage()
+  }
+
+  And("""^(?:I )?choose radio option (.*) for You have added (.+) authorisation page$""") {
```
Comment (vajarova) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:47]: As @JohnWhitfield comment 
needs an 's?' after the 'n'.
so makes it 'authorisations?'

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    (answer: String, numberOfAuthorisations: String) =>
+      AddAnotherAuthorisationPage
+        .loadPage(numberOfAuthorisations)
+        .select(answer)
+        .submitPage()
+  }
+
+  When("""^(?:I )?choose to click on the (.*) link on You have added (.+) authorisations page$""") {
```
Comment (vajarova) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:55]: as per @JohnWhitfield comment authorisations needs a '?'

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  And("""^(?:I )?choose radio option (.+) for Are you sure you want to remove (.*) page$""") {
+    (answer: String, typeOfAuthorisation: String) =>
+      RemoveAnAuthorisationPage
+        .loadPage(typeOfAuthorisation)
+        .select(answer)
+        .submitPage()
+  }
+
+  And("""^(?:I )?enter date on the limit date page$""") { () =>
```
Comment (vajarova) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:70]: add the before date as pet @JohnWhitfield comment

**Review by vajarova — CHANGES_REQUESTED** [review]

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala (hunk 1 above)
Comment (VijayKumarTech) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:47]: done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala (hunk 2 above)
Comment (VijayKumarTech) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:55]: done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala (hunk 3 above)
Comment (VijayKumarTech) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Departures/Transport/AuthorisationsStepDef.scala:70]: done

**Comment by vajarova** [comment]
Superseded by https://github.com/hmrc/manage-transit-movements-ui-journey-tests/pull/82

**Pull request closed without merging**